src/transaction.rs

```diff
@@ -0,0 +1,254 @@
+use crate::{
+    ColumnFamily, CreateIter, DBRawIterator, DBVector, Error, OptimisticTransactionDB, ReadOptions,
+};
+use ffi;
+use libc::{c_char, size_t};
+use std::marker::PhantomData;
+
+pub struct Transaction<'a> {
+    pub(crate) inner: *mut ffi::rocksdb_transaction_t,
+    pub(crate) db: PhantomData<&'a OptimisticTransactionDB>,
+    snapshot: Option<*const ffi::rocksdb_snapshot_t>,
+}
+
+impl<'a> Transaction<'a> {
+    pub(crate) fn new(
+        inner: *mut ffi::rocksdb_transaction_t,
+        snapshot: Option<*const ffi::rocksdb_snapshot_t>,
+    ) -> Transaction<'a> {
+        Transaction {
+            inner,
+            db: PhantomData,
+            snapshot,
+        }
+    }
+
+    /// commits a transaction
+    pub fn commit(&self) -> Result<(), Error> {
+        unsafe {
+            ffi_try!(ffi::rocksdb_transaction_commit(self.inner,));
+        }
+        Ok(())
+    }
+
+    /// Delete a key inside a transaction
+    ///
+    /// ColumnFamilyHandle: default
+    pub fn delete(&self, key: &[u8]) -> Result<(), Error> {
```

Comment:
Can you make `key` generic? eg: `K: AsRef<[u8]>` See #253 for some examples of similar changes.